sample-apps/donet-sample-app/Controllers/MetricEmitter.cs

```diff
@@ -119,18 +144,25 @@ public MetricEmitter()
 
         public void updateTotalHeapSizeMetric(int heapSize) {
             totalHeapSize += heapSize;
+            //totalHeapSizeObserver.Publish(totalheap);
         }
 
-        public void updateTotalThreadSizeMetric(int threadSize) {
-            totalThreadSize += threadSize;
-        }
+        public void updateTotalThreadSizeMetric(int totalthreads) {
+            totalThreadsObserver.Add(totalthreads,
+                new KeyValuePair<string, object>("signal", "metric"),
+                new KeyValuePair<string, object>("language", "dotnet"),
+                new KeyValuePair<string, object>("metricType", "request"));
+        }   
 
         public void updateCpuUsageMetric(int cpuUsage) {
-            cpuUsage = cpuUsage;
+            cpuUsage += cpuUsage;
```

Comment:
```suggestion
            this.cpuUsage += cpuUsage;
```
Is specifying with `this` required or should we just change the function parameter to be more clear on scoping?

Reply:
I agree on using `this` to scope the variable to the current instance.